### PR TITLE
[CI] Restore uploading backend image to ECR as part of the release workflow.

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
It was mistakenly removed, along with the intended removal of the GUI cloud upload.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -53,6 +53,65 @@ jobs:
     outputs:
       ENSO_RELEASE_ID: ${{ steps.prepare.outputs.ENSO_RELEASE_ID }}
       ENSO_VERSION: ${{ steps.prepare.outputs.ENSO_VERSION }}
+  enso-build-ci-gen-job-deploy-runtime-linux-x86_64:
+    name: Upload Runtime to ECR (linux, x86_64)
```
Comment (PabloBuchu) [inline .github/workflows/release.yml:57]: maybe `Upload Runtime to Cloud AWS ECR (linux, x86_64)`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -53,6 +53,65 @@ jobs:
...
+        name: Setup conda (GH runners only)
+        uses: s-weigand/setup-conda@v1.2.1
+        with:
+          update-conda: false
+          conda-channels: anaconda, conda-forge
+      - if: startsWith(runner.name, 'GitHub Actions') || startsWith(runner.name, 'Hosted Agent')
+        name: Installing wasm-pack
+        uses: jetli/wasm-pack-action@v0.4.0
+        with:
+          version: v0.10.2
```
Comment (PabloBuchu) [inline .github/workflows/release.yml:75]: do we still need wasm-pack?

Comment (mwu-tow) [reply .github/workflows/release.yml:75]: We do still have WASM parser, the setup routine is shared across the jobs.

**Review by PabloBuchu — COMMENTED** [review]
see the comments

File: .github/workflows/release.yml (hunk 1 above)
Comment (mwu-tow) [reply .github/workflows/release.yml:57]: Unless you find the name misleading, I'd rather not make it any longer. It is already the longest name in the workflow and horizontal space for job names is rather limited. Making it longer will just cause GitHub to trim it when displaying in the navigation bar.
In the job graph it is already trimmed to 
obraz

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**